Lib/ufo2ft/otfPostProcessor.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function, division, absolute_import, unicode_literals
+
+import tempfile
+
+from compreffor import Compreffor
+from fontTools.ttLib import TTFont
+
+
+class OTFPostProcessor(object):
+    """Does some post-processing operations on a compiled OpenType font, using
+    info from the source UFO where necessary.
+    """
+
+    def __init__(self, otf, ufo):
+        self.ufo = ufo
+        tmp_file = tempfile.NamedTemporaryFile()
+        tmp_path = tmp_file.name
+        otf.save(tmp_path)
+        self.otf = TTFont(tmp_path)
+
+    def process(self, useProductionNames=True, optimizeCff=True):
+        if useProductionNames:
+            self._rename_glyphs_from_ufo()
+        if optimizeCff and 'CFF ' in self.otf:
+            comp = Compreffor(self.otf)
+            comp.compress()
+        return self.otf
+
+    def _rename_glyphs_from_ufo(self):
+        """Rename glyphs using glif.lib.public.postscriptNames in UFO."""
+
+        rename_map = {
+            g.name: self._build_production_name(g) for g in self.ufo}
+        rename = lambda names: [rename_map[n] for n in names]
+
+        self.otf.setGlyphOrder(rename(self.otf.getGlyphOrder()))
+        if 'CFF ' in self.otf:
+            cff = self.otf['CFF '].cff.topDictIndex[0]
+            char_strings = cff.CharStrings.charStrings
+            cff.CharStrings.charStrings = {
+                rename_map.get(n, n): v for n, v in char_strings.items()}
+            cff.charset = rename(cff.charset)
+
+    def _build_production_name(self, glyph):
+        """Build a production name for a single glyph."""
+
+        # use name from Glyphs source if available
+        production_name = glyph.lib.get('public.postscriptName')
+        if production_name:
+            return production_name
+
+        # use name derived from unicode value
+        unicode_val = glyph.unicode
+        if glyph.unicode is not None:
+            return '%s%04X' % (
+                'u' if unicode_val > 0xffff else 'uni', unicode_val)
```

Comment:
> Glyphs for example will only use AGL names for characters through latin extended-A, and uni* for most everything else.

That sounds about right, AGLFN only covers some latin charset (unlike AGL, but fonttools.agl contains the AGLFN as that's what should be used for new fonts). 
Per Adobe spec what's outside this list should be computed with uniXXXX rules.

(Previously with the AGL they maintained a full glyph list for everything but it was a mess with backwards-incompatible changes etc. see e.g. [here](http://unicode.org/mail-arch/unicode-ml/y2002-m12/0234.html) so they froze AGLFN you put a generic naming scheme for the rest, afaict).